cmd/power/power.go

```diff
@@ -19,6 +21,53 @@ const (
 	DOCKER_CGROUPS_FILE = "/proc/self/cgroup"
 )
 
+func PowerMain() {
+	app := cli.NewApp()
+
+	app.Name = os.Args[0]
+	app.Version = config.VERSION
+	app.Author = "Rancher Labs, Inc."
+	app.Email = "sid@rancher.com"
+	app.EnableBashCompletion = true
+	app.Action = power
+	app.HideHelp = true
+	app.Flags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "f",
+			Usage: "Does not invoke shutdown, and instead performs the actual action",
+		},
+		cli.BoolFlag{
+			Name:  "n",
+			Usage: "Does not call sync before bringing down system",
+		},
+		cli.BoolFlag{
+			Name:  "P",
+			Usage: "Halt behaves like poweroff",
+		},
+		cli.BoolFlag{
+			Name:  "i",
+			Usage: "Brings down configured network devices before shutdown",
+		},
+		cli.BoolFlag{
+			Name:  "H",
+			Usage: "Brings down IDE drives before shutdown",
+		},
+	}
+	app.Run(os.Args)
+}
+
+func power(c *cli.Context) {
+	force := c.Bool("f")
+	switch os.Args[0] {
+	case "poweroff":
```

Comment:
Args[0] may equal the full path of the file also, not merely the executable name.  For example "/sbin/poweroff"